dotCMS/src/integration-test/java/com/dotmarketing/business/VersionableAPITest.java

```diff
@@ -0,0 +1,331 @@
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.IntegrationTestInitService;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.portal.model.User;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Created by Erick Gonzalez
+ */
+public class VersionableAPITest {
+	
+	private static User user;
+	private static Host host;
+	
+	@BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+        
+        user = APILocator.getUserAPI().getSystemUser();
+        host = APILocator.getHostAPI().findDefaultHost(user, false);
+	}
+
+	@Test
+	public void testFindWorkingVersionHTMLPage() throws Exception{
+		//Create HTMLPage
+		String ext="."+Config.getStringProperty("VELOCITY_PAGE_EXTENSION");
+		
+		Template template=new Template();
+        template.setTitle("TEST TEMPLATE VERSIONABLE");
+        template.setBody("<html><body> I'm mostly empty </body></html>");
+        template=APILocator.getTemplateAPI().saveTemplate(template, host, user, false);
+        
+        Folder folder = APILocator.getFolderAPI().createFolders("/testingVersionable", host, user, false);
+		
+		HTMLPage page=new HTMLPage();
+        page.setPageUrl("testpage"+ext);
+        page.setFriendlyName("testpage"+ext);
+        page.setTitle("testpage"+ext);
+        page=APILocator.getHTMLPageAPI().saveHTMLPage(page, template, folder, user, false);
+        
+        //Call Versionable
+        Versionable verAPI = APILocator.getVersionableAPI().findWorkingVersion(page.getIdentifier(), user, false);
+        
+        //Check Same HTMLPage
+        assertEquals(verAPI.getTitle(),page.getTitle());
+        assertEquals(verAPI.getInode(),page.getInode());
+        
+        //Delete Template, Folder, HTMLPage
+        APILocator.getHTMLPageAPI().delete(page, user, false);
+        APILocator.getFolderAPI().delete(folder, user, false);
+        APILocator.getTemplateAPI().delete(template, user, false);
+	}
+	
+	@Test
+	public void testFindWorkingVersionContainer() throws Exception{
+		//Create Container
+		Container c = new Container();
+        c.setFriendlyName("Versionable Container Working");
+        c.setTitle("Versionable Container Working");
+        c.setMaxContentlets(2);
+        c.setPreLoop("preloop code");
+        c.setPostLoop("postloop code");
+
+        Structure st=CacheLocator.getContentTypeCache().getStructureByVelocityVarName("host");
+        List<ContainerStructure> csList = new ArrayList<ContainerStructure>();
+        ContainerStructure cs = new ContainerStructure();
+        cs.setStructureId(st.getInode());
+        cs.setCode("this is the code");
+        csList.add(cs);
```

Comment:
Remember you can use Collections.singletonList(something) for lists with only one item